axelrod/strategies/mindreader.py

```diff
@@ -46,15 +46,25 @@ class MindReader(Player):
     """A player that looks ahead at what the opponent will do and decides what to do."""
 
     name = 'Mind Reader'
-    stochastic = True # Don't cache me
+    behaviour = {
+        'memory_depth': -10,
+        'stochastic': False,
```

Comment:
> I'm not sure if it actually inspects the source -- it doesn't explicitly read the source with inspect, but it does use the source to simulate. Let's be sure that the docs explain that both behaviors are included.

I agree, wasn't too sure what to do without creating another dimension (which I would not like to do). I suggest leaving it as it is but documenting? Or is it just better to leave just having `manipulates_opponents_state` which arguably is a nicer way of putting it...